### PR TITLE
add -test.skip

### PR DESCRIPTION
**Pull request by dgryski**
Should have just pulled in `match.go` from the beginning..

### CODE REVIEW

**Comment by dgryski** [comment]
Bah. Need import updated tests too.

File: main.go [hunk 1]
```diff
@@ -1427,6 +1430,7 @@ func main() {
 		flag.BoolVar(&testConfig.Verbose, "v", false, "verbose: print additional output")
 		flag.BoolVar(&testConfig.Short, "short", false, "short: run smaller test suite to save time")
 		flag.StringVar(&testConfig.RunRegexp, "run", "", "run: regexp of tests to run")
+		flag.StringVar(&testConfig.SkipRegexp, "skip", "", "run: regexp of tests to run")
```
Comment (aykevl) [inline main.go]: Copy-paste bug:
```suggestion
		flag.StringVar(&testConfig.SkipRegexp, "skip", "", "run: regexp of tests to skip")
```

**Review by aykevl — COMMENTED** [review]
Looks good to me.

I assume match.go is mostly from upstream? I didn't check it thoroughly, but if it is basically a copy from upstream Go I'll trust it is correct.

**Comment by dgryski** [comment]
Yes. Match had previously been imported from upstream. Now it's a newer version with skip support.

The only change is the bare metal fake matcher, which I moved out of that file to make updating easier in the future.

**Comment by dgryski** [comment]
PTAL

**Review by aykevl — APPROVED** [review]

**Pull request merged**